tfe/resource_tfe_team_access.go

```diff
@@ -0,0 +1,124 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFETeamAccess() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFETeamAccessCreate,
+		Read:   resourceTFETeamAccessRead,
+		Delete: resourceTFETeamAccessDelete,
+
+		Schema: map[string]*schema.Schema{
+			"access": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+					switch v.(string) {
+					case "admin", "read", "write":
+						// These are the two valid options, so we return without errors.
+						return nil, nil
+					default:
+						return nil, []error{fmt.Errorf(
+							"%q must be either 'admin', 'read' or 'write'", k),
+						}
+					}
+				},
+			},
+
+			"team_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"workspace_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFETeamAccessCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get access, team ID, workspace and organization.
+	access := d.Get("access").(string)
+	teamID := d.Get("team_id").(string)
+	workspace, organization := unpackWorkspaceID(d.Get("workspace_id").(string))
+
+	// Get the team.
+	tm, err := tfeClient.Teams.Read(ctx, teamID)
+	if err != nil {
+		return fmt.Errorf("Error retrieving team %s: %v", teamID, err)
+	}
+
+	// Get the workspace.
+	ws, err := tfeClient.Workspaces.Read(ctx, organization, workspace)
+	if err != nil {
+		return fmt.Errorf(
+			"Error retrieving workspace %s from organization %s: %v", workspace, organization, err)
+	}
+
+	// Create a new options struct.
+	options := tfe.TeamAccessAddOptions{
+		Access:    tfe.Access(tfe.AccessType(access)),
+		Team:      tm,
+		Workspace: ws,
+	}
+
+	log.Printf("[DEBUG] Give team %s %s access to workspace: %s", tm.Name, access, ws.Name)
+	tmAccess, err := tfeClient.TeamAccess.Add(ctx, options)
+	if err != nil {
+		return fmt.Errorf(
+			"Error giving team %s %s access to workspace %s: %v", tm.Name, access, ws.Name, err)
+	}
+
+	d.SetId(tmAccess.ID)
+
+	return resourceTFETeamAccessRead(d, meta)
+}
+
+func resourceTFETeamAccessRead(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	log.Printf("[DEBUG] Read configuration of team access: %s", d.Id())
+	tmAccess, err := tfeClient.TeamAccess.Read(ctx, d.Id())
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			log.Printf("[DEBUG] Team access %s does no longer exist", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading configuration of team access %s: %v", d.Id(), err)
+	}
+
+	// Update config.
+	d.Set("access", string(tmAccess.Access))
+	d.Set("team_id", tmAccess.Team.ID)
```

Comment:
is `Team` a pointer? if so, what happens if this is nil?

Reply:
You know what happens then 😏It would be a serious bug in the TFE API if this value was not returned, but to safeguard us here I added a check 👍 

Fixed in the next commit.